include/envoy/compressor/compressor.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+
+namespace Envoy {
+namespace Compressor {
+
+/**
+ * Public compressor allows compress and decompress buffer data.
+ */
+class Compressor {
+public:
+  virtual ~Compressor() {}
+
+  /**
+   * Compress data from one buffer into another buffer
+   * @param in supplies the buffer with data to be compressed
+   * @param out supplies the buffer to output compressed data
+   * @return bool indicating whether operation succeeded or not.
+   */
+  virtual bool compress(Buffer::Instance& in, Buffer::Instance& out) PURE;
+
+  /**
+   * Decompress data from one buffer into another buffer
+   * @param in supplies the buffer with data to be decompressed
+   * @param out supplies the buffer to output uncompressed data
+   * @return bool indicating whether operation succeeded or not.
+   */
+  virtual bool decompress(Buffer::Instance& in, Buffer::Instance& out) PURE;
+
+  /**
+   * Ends compression/decompression.
+   * @return bool indicating whether operation succeeded or not.
+   */
+  virtual bool finish() PURE;
```

Comment:
finish() is definitely not needed here. I added because in old chat with Matt he suggested me something similar, so I thought it could be some sort of a common operation for compressors in general. I removed it anyways.